### DISCUSSION TITLE
How to realize the Robin Boundary condition on pde

### DISCUSSION THREAD

**Discussion by jsfeng-fudan** [LLNL/serac#756] [General]
I have one 1d ode :
       -y' ' = 3 in domain x in [0,1], with the boundary 
       y'[0]-y[0] = -1 and    y'[0]+y[0] = +1; 
I tried to solve this problem using mfem library. but I can not arrive at correct solution.
The correct solution is 
![图片](https://user-images.githubusercontent.com/101965017/181595637-fe3b37db-c9b7-4e42-b60e-fc0360d4261f.png)

my solution is 
![图片](https://user-images.githubusercontent.com/101965017/181595530-ab7983e1-2d39-43cb-93f9-9d7da5f22afc.png)

I post my code for enforce the Robin boundary ..

        rbc_bdr = 1;
        fespace.GetEssentialTrueDofs(rbc_bdr, ess_tdof_list);
       
        double rbc_a_val1 = -1.0; // for left endpoint
        double rbc_b_val1 = -1.0; 
        double rbc_a_val2 = 1.0;  // for right endpoint
        double rbc_b_val2 = 1.0; 
        ConstantCoefficient rbc_a_coeff1(rbc_a_val1);// entering a matrix
        ConstantCoefficient rbc_a_coeff2(rbc_a_val2);// entering a matrix
        ConstantCoefficient rbc_b_coeff1(rbc_b_val1); // entering b linear form
        ConstantCoefficient rbc_b_coeff2(rbc_b_val2); // entering b linear form           
        b.AddDomainIntegrator(new DomainLFIntegrator(f_src));
        // key steps: add B.C. into linear system
        // Add Robin BCs n.(matCoef Grad u) + rbcACoef u = rbcBCoef on the boundary
        // // marked in the rbc_marker array.
        // left
        rbc_bdr = 0;
        rbc_bdr[0] =1 ;
        b.AddBoundaryIntegrator(new BoundaryLFIntegrator(rbc_b_coeff1), rbc_bdr);
        rbc_bdr = 0;
        rbc_bdr[1] =1 ;
        b.AddBoundaryIntegrator(new BoundaryLFIntegrator(rbc_b_coeff2), rbc_bdr);                                   
        b.Assemble();
        //getchar();
        a.AddDomainIntegrator( new DiffusionIntegrator(one));
        // // Add Robin BCs n.(matCoef grad u) + rbcACoef u = rbcBCoef on the boundary
        // // marked in the rbc_marker array.
        rbc_bdr = 0;
        rbc_bdr[0] = 1;
        a.AddBoundaryIntegrator(new MassIntegrator(rbc_a_coeff1), rbc_bdr);
        rbc_bdr = 0;
        rbc_bdr[1] = 1;
        a.AddBoundaryIntegrator(new MassIntegrator(rbc_a_coeff2), rbc_bdr);       
        cout << a.Size() << endl;
        a.Assemble();

**Comment by jsfeng-fudan** [comment]
sorry,   Robin boundary condition reads :
y'[0]-y[0] = -1 and y'[1]+y[1] = +1;
Can someone help to tackle this problem.

**Comment by jamiebramwell** [comment]
As this problem only involves MFEM data structures, I suggest posting in their repository: https://github.com/mfem/mfem/
